lighty-modules/lighty-gnmi/lighty-gnmi-test/src/test/java/io/lighty/modules/gnmi/test/gnmi/GnmiWithoutRestconfTest.java

```diff
@@ -0,0 +1,497 @@
+    private static final QName ADD_KEYSTORE_RPC_QN = QName.create(CERT_STORAGE_QN_MODULE, "add-keystore-certificate");
+    private static final QName ADD_KEYSTORE_INPUT_QN = QName.create(ADD_KEYSTORE_RPC_QN, "input");
+    private static final QName KEYSTORE_ID_QN = QName.create(CERT_STORAGE_QN_MODULE, "keystore-id");
+    private static final QName CA_CERT_QN = QName.create(CERT_STORAGE_QN_MODULE, "ca-certificate");
+    private static final QName CLIENT_KEY_QN = QName.create(CERT_STORAGE_QN_MODULE, "client-key");
+    private static final QName PASSPHRASE_QN = QName.create(CERT_STORAGE_QN_MODULE, "passphrase");
+    private static final QName CLIENT_CERT_QN = QName.create(CERT_STORAGE_QN_MODULE, "client-cert");
+
+    private static final QNameModule YANG_STORAGE_QN_MODULE
+            = QNameModule.create(URI.create("urn:lighty:gnmi:yang:storage"), Revision.of("2021-03-31"));
+    private static final QName UPLOAD_YANG_RPC_QN = QName.create(YANG_STORAGE_QN_MODULE, "upload-yang-model");
+    private static final QName UPLOAD_YANG_INPUT_QN = QName.create(UPLOAD_YANG_RPC_QN, "input");
+    private static final QName GNMI_YANG_MODELS_QN = QName.create(YANG_STORAGE_QN_MODULE, "gnmi-yang-models");
+    private static final QName GNMI_YANG_MODEL_QN = QName.create(GNMI_YANG_MODELS_QN, "gnmi-yang-model");
+    private static final QName YANG_NAME_QN = QName.create(GNMI_YANG_MODEL_QN, "name");
+    private static final QName YANG_VERSION_QN = QName.create(GNMI_YANG_MODEL_QN, "version");
+    private static final QName YANG_BODY_QN = QName.create(GNMI_YANG_MODEL_QN, "body");
+
+
+    private static LightyController lightyController;
+    private static GnmiSouthboundModule gnmiSouthboundModule;
+    private static SimulatedGnmiDevice gnmiDevice;
+
+
+    @BeforeAll
+    public static void startUp() throws ConfigurationException, ExecutionException, InterruptedException, IOException,
+            InvalidAlgorithmParameterException, NoSuchPaddingException, NoSuchAlgorithmException,
+            InvalidKeySpecException, InvalidKeyException {
+
+        lightyController = new LightyControllerBuilder()
+                .from(ControllerConfigUtils.getConfiguration(Files.newInputStream(CONFIGURATION_PATH)))
+                .build();
+        lightyController.start().get();
+
+        gnmiSouthboundModule = new GnmiSouthboundModuleBuilder()
+                .withConfig(GnmiConfigUtils.getGnmiConfiguration(Files.newInputStream(CONFIGURATION_PATH)))
+                .withLightyServices(lightyController.getServices())
+                .withExecutorService(Executors.newCachedThreadPool())
+                .withEncryptionService(createEncryptionService())
+                .build();
+        gnmiSouthboundModule.start().get();
+
+        gnmiDevice = getUnsecureGnmiDevice(DEVICE_ADDRESS, DEVICE_PORT);
+        gnmiDevice.start();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
```

Comment:
@ AfterAll should not throw exception - try to release all the resources even if some release failed

Reply:
done